From d808bc0bb0d06ce510a8083e2b16111b97bbc881 Mon Sep 17 00:00:00 2001
From: Jingtao Chen <jingtaoc@codeaurora.org>
Date: Thu, 31 Mar 2016 12:24:23 +0800
Subject: [PATCH 20/31] APBU-11786 bugfix

Avoid simply extend stream duration when segment stop time
is later than stream duration.

Change-Id: I0ccd6e66426b0019881b0f0299b7cee3e7798813
Signed-off-by: Jingtao Chen <jingtaoc@codeaurora.org>
---
 gst/isomp4/qtdemux.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index d727a54..49195fb 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -7905,7 +7905,11 @@ qtdemux_parse_segments (GstQTDemux * qtdemux, QtDemuxStream * stream,
             " past the end of the file duration %" GST_TIME_FORMAT
             " it will be truncated", i, GST_TIME_ARGS (segment->stop_time),
             GST_TIME_ARGS (qtdemux->segment.stop));
-        qtdemux->segment.stop = segment->stop_time;
+        /* Jingtao: should not simply extend the stream duration because
+        it will cause in duration dismatch sometimes. I've no idea about
+        the explicit reason but I guess the stream may be not correctly
+        labeled */
+        //qtdemux->segment.stop = segment->stop_time;
       }
     }
     GST_DEBUG_OBJECT (qtdemux, "found %d segments", count);
-- 
1.9.1

